test/testcafe/framework/page-objects/UiDemoPageObject.js

```diff
@@ -0,0 +1,33 @@
+import BasePageObject from './BasePageObject';
+import { ClientFunction } from 'testcafe';
+
+
+export default class UiDemoPageObject extends BasePageObject {
+  constructor(t) {
+    super(t);
+  }
+
+  async scrollToNextPage() {
+    const getClientHeight = ClientFunction(() => window.document.documentElement.clientHeight);
+    const windowheight = await getClientHeight();
+    await this.t.scrollBy(0, windowheight);
+  }
+
+  async hideProgressBar() {
+    await ClientFunction(() => {
+      // the testcafe progress bar element is in the shadow dom so we cannot query it directly
+      const style = document.createElement('style');
+      style.textContent = '.progress-bar-hammerhead-shadow-ui { visibility: hidden !important; }';
+      document.head.appendChild(style);
+    })();
+  }
+
+  async stopSpinnerAnimation() {
+    await ClientFunction(() => {
+      const spinnerRoot = document.querySelector('.MuiCircularProgress-root');
+      const spinnerCircle = document.querySelector('.MuiCircularProgress-circle');
+      spinnerRoot.setAttribute('style', 'animation: none;');
+      spinnerCircle.setAttribute('style', 'animation: none;');
```

Comment:
I know you mentioned you were having issues with getting the Spinner to freeze yesterday. I just tried it locally and was able to get it to freeze by using `animation: none!important`. I know `!important` isn't great style, but I think it's fine for this use case, have you tried using it here?

Reply:
Hmm I wasn't able to get it to work using that.  The spinner appears in your screenshot?

Reply:
was able to get it to work by appending the style to the `head` instead